snakemake/executors/slurm/slurm_submit.py

```diff
@@ -0,0 +1,276 @@
+        has the internal handling via job.resources
+        """
+        return job.dynamic_wildcards.copy()
+
+    def run(self, job, callback=None, submit_callback=None, error_callback=None):
+        super()._run(job)
+        workdir = os.getcwd()
+        jobid = job.jobid
+        # generic part of a submission string:
+
+        os.makedirs(".snakemake/slurm-logs", exist_ok=True)
+
+        try:
+            call = "sbatch -A {account} -p {partition} \
+                    -J {jobname} \
+                    -o .snakemake/slurm_logs/%x_%j.log \
+                    --export=ALL".format(
+                **job.resources, jobname=self.get_jobname(job)
+            )
+        except KeyError as e:
+            # TODO: make explicit message for account / partition
+            logger.error(
+                "Missing job submission key '{}' for job '{}'.".format(
+                    e.args[0], job.name
+                )
+            )
+            sys.exit(1)
+
+        if not job.resources.get("walltime_minutes"):
+            logger.warning(
+                "No wall time limit is set, setting 'walltime_minutes' to 1."
+            )
+        call += " -t {walltime_minutes}".format(
+            walltime_minutes=job.resources.get("walltime_minutes", default_value=1)
+        )
+        if job.resources.get("constraint"):
+            call += " -C {constraint}".format(**job.resources)
+        # TODO: implement when tempfs-resource is defined
+        # if job.resources.get()
+        if job.resources.get("mem_mb_per_cpu"):
+            call += " --mem-per-cpu={mem_mb_per_cpu}".format(**job.resources)
+        elif job.resources.get("mem_mb"):
+            call += " --mem {mem_mb}".format(**job.resources)
+        else:
+            logger.warning(
+                "No job memory information ('mem_mb' or 'mem_mb_per_cpu') is given - submitting without. This might or might not work on your cluster."
+            )
+
+        exec_job = self.format_job_exec(job)
+
+        # MPI job
+        if job.resources.get("mpi", False):
+            if job.resources.get("nodes", False):
+                call += " --nodes={}".format(job.resources.get("nodes", 1))
+            if job.resources.get("tasks", False):
+                call += " --ntasks={}".format(job.resources.get("tasks", 1))
+            if job.resources.get("threads", False):
+                call += " -c={}".format(job.resources.get("threads", 1))
+        # ordinary smp application
+        elif not job.is_group():
+            # TODO: this line will become longer
+            # TODO: hence the single command, yet
+            if job.threads == 0:
+                call += f" -n 1 -c 1 {exec_job}"
+            else:
+                call += f" -n 1 -c {job.threads}"
+        else:
+            ntasks = max(map(len, job.toposorted))
+            threads = sum(j.threads for j in job)
+            call += f" -n {ntasks} -c {threads}"
```

Comment:
I'm likely confused here, but it looks like you're adding up threads across all jobs in a group job then supplying that as the number of cpus *per* task. So 50 jobs at one thread each would request 50 cores per task. Whereas I think you'd want 50 tasks with one core per task

Reply:
Yes, you are right: I think that `max()` instead of `sum()` is to be used, here - that would be a price to pay: less synchronization overhead (perhaps even less I/O), but some in-efficiency instead. I could live with that and changed the code accordingly.

However, this still needs testing! Thank you for spotting this glitch!

Due to a public holiday (and a family field trip) I will continue working on it next Monday.